urdf_model/include/urdf_model/pose.h

```diff
@@ -166,13 +166,27 @@ class Rotation
 
   void clear() { this->x=this->y=this->z=0.0;this->w=1.0; }
 
+  /**
+   * \brief Normalize the elements of this quaternion
+   *
+   * Note: This function historically checked if the norm of the elements
+   * equaled 0 using a strict floating point equals (s == 0.0) to prevent a
+   * divide by zero error. This comparison will cause the compiler to issue a
+   * warning with '-Wfloat-equal'. Comparing against a small epsilon would
+   * likely be more ideal, but its choice may be application specific and could
+   * change behavior of legacy code if chosen poorly. Using <= handles the
+   * warning without changing the behavior of this function, even though it's
+   * less than ideal.
```

Comment:
Nit:

```suggestion
   * change behavior of legacy code if chosen poorly. Using <= silences the
   * warning without changing the behavior of this function, even though there
   * are no situations in which `squared_norm` can be less than 0.0.
```

I'd also move this comment to be right above the `if (squared_norm <= 0.0)` check.

Reply:
Done and done